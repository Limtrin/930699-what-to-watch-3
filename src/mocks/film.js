export const Film = {
  id: `grand-budapest`,
  name: `The Grand Budapest Hotel`,
  genre: `Dramas`,
  year: 2019,
  image: `img/bohemian-rhapsody.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `2h 22m`,
  rating: 9.5,
  votes: 123,
  director: `Frank Spaskukotski`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
  reviews: [
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    }
  ],
  starring: [
    `Tim Robbins`,
    `Morgan Freeman`,
    `Bob Gunton`,
    `William Sadler`,
    `Clancy Brown`,
    `Gil Bellows`,
    `James Whitmore`
  ],
};
