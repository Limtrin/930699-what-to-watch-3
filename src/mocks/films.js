export const FilmsList = [
  {
    id: `bohemian-rhapsody`,
    name: `Bohemian Rhapsody`,
    genre: `Dramas`,
    year: 2019,
    image: `img/bohemian-rhapsody.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  },
  {
    id: `aviator`,
    name: `Aviator`,
    genre: `Dramas`,
    year: 2010,
    image: `img/aviator.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 12m`,
    rating: 8.5,
    votes: 500,
    director: `Frank Zappa`,
    description: `Zero, a junior lobby boy, becomes Gustaves friend and protege.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `John Travolta`,
      `Samuel L. Jackson`,
      `Uma Thurman`,
      `Harvey Keitel`,
      `Tim Roth`,
      `Amanda Plummer`,
      `Maria de Medeiros`,
      `Ving Rhames`,
      `Eric Stoltz`,
      `Rosanna Arquette`,
      `Christopher Walken`,
      `Bruce Willis`
    ],
  },
  {
    id: `johnny-english`,
    name: `Johnny English`,
    genre: `Comedies`,
    year: 2013,
    image: `img/johnny-english.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 00m`,
    rating: 7.5,
    votes: 1000,
    director: `Frank Iero`,
    description: `Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Anthony Perkins`,
      `Vera Miles`,
      `John Gavin`,
      `Martin Balsam`,
      `John McIntire`,
      `Janet Leigh`
    ],
  },
  {
    id: `shutter-island`,
    name: `Shutter Island`,
    genre: `Documentary`,
    year: 2010,
    image: `img/shutter-island.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 23m`,
    rating: 9.2,
    votes: 1800,
    director: `Steven Spilberg`,
    description: `When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Christian Bale`,
      `Michael Caine`,
      `Heath Ledger`,
      `Gary Oldman`,
      `Aaron Eckhart`,
      `Maggie Gyllenhaal`,
      `Morgan Freeman`
    ],
  },
  {
    id: `snatch`,
    name: `Snatch`,
    genre: `Comedies`,
    year: 2009,
    image: `img/snatch.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ], starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`,
      `Hugo Weaving`,
      `Joe Pantoliano`
    ],
  },
  {
    id: `war-of-the-worlds`,
    name: `War of the Worlds`,
    genre: `Thrillers`,
    year: 2008,
    image: `img/war-of-the-worlds.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 12m`,
    rating: 8.5,
    votes: 500,
    director: `Frank Zappa`,
    description: `Zero, a junior lobby boy, becomes Gustaves friend and protege.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`,
      `Hugo Weaving`,
      `Joe Pantoliano`
    ],
  },
  {
    id: `mindhunter`,
    name: `Mindhunter`,
    genre: `Dramas`,
    year: 2015,
    image: `img/mindhunter.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 00m`,
    rating: 7.5,
    votes: 1000,
    director: `Frank Iero`,
    description: `Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tom Skerritt`,
      `Sigourney Weaver`,
      `Veronica Cartwright`,
      `Harry Dean Stanton`,
      `John Hurt`,
      `Ian Holm`,
      `Yaphet Kotto`
    ],
  },
  {
    id: `orlando`,
    name: `Orlando`,
    genre: `Dramas`,
    year: 2011,
    image: `img/orlando.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 00m`,
    rating: 7.5,
    votes: 1000,
    director: `Ice Loched`,
    description: `Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tom Skerritt`,
      `Sigourney Weaver`,
      `Veronica Cartwright`,
      `Harry Dean Stanton`,
      `John Hurt`,
      `Ian Holm`,
      `Yaphet Kotto`
    ],
  },
  {
    id: `snatch2`,
    name: `Snatch`,
    genre: `Comedies`,
    year: 2009,
    image: `img/dardjeeling-limited.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ], starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`,
      `Hugo Weaving`,
      `Joe Pantoliano`
    ],
  },
  {
    id: `fantastic-beasts-the-crimes-of-grindelwald`,
    name: `Bohemian Rhapsody`,
    genre: `Dramas`,
    year: 2019,
    image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  },
  {
    id: `midnight-special`,
    name: `Aviator`,
    genre: `Dramas`,
    year: 2010,
    image: `img/midnight-special.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 12m`,
    rating: 8.5,
    votes: 500,
    director: `Frank Zappa`,
    description: `Zero, a junior lobby boy, becomes Gustaves friend and protege.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `John Travolta`,
      `Samuel L. Jackson`,
      `Uma Thurman`,
      `Harvey Keitel`,
      `Tim Roth`,
      `Amanda Plummer`,
      `Maria de Medeiros`,
      `Ving Rhames`,
      `Eric Stoltz`,
      `Rosanna Arquette`,
      `Christopher Walken`,
      `Bruce Willis`
    ],
  },
  {
    id: `moonrise-kingdom`,
    name: `Johnny English`,
    genre: `Comedies`,
    year: 2013,
    image: `img/moonrise-kingdom.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 00m`,
    rating: 7.5,
    votes: 1000,
    director: `Frank Iero`,
    description: `Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Anthony Perkins`,
      `Vera Miles`,
      `John Gavin`,
      `Martin Balsam`,
      `John McIntire`,
      `Janet Leigh`
    ],
  },
  {
    id: `pulp-fiction`,
    name: `Shutter Island`,
    genre: `Dramas`,
    year: 2010,
    image: `img/pulp-fiction.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 23m`,
    rating: 9.2,
    votes: 1800,
    director: `Steven Spilberg`,
    description: `When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Christian Bale`,
      `Michael Caine`,
      `Heath Ledger`,
      `Gary Oldman`,
      `Aaron Eckhart`,
      `Maggie Gyllenhaal`,
      `Morgan Freeman`
    ],
  },
  {
    id: `revenant`,
    name: `Snatch`,
    genre: `Dramas`,
    year: 2009,
    image: `img/revenant.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ], starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`,
      `Hugo Weaving`,
      `Joe Pantoliano`
    ],
  },
  {
    id: `seven-years-in-tibet`,
    name: `War of the Worlds`,
    genre: `Dramas`,
    year: 2008,
    image: `img/seven-years-in-tibet.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 12m`,
    rating: 8.5,
    votes: 500,
    director: `Frank Zappa`,
    description: `Zero, a junior lobby boy, becomes Gustaves friend and protege.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`,
      `Hugo Weaving`,
      `Joe Pantoliano`
    ],
  },
  {
    id: `we-need-to-talk-about-kevin`,
    name: `Mindhunter`,
    genre: `Dramas`,
    year: 2015,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 00m`,
    rating: 7.5,
    votes: 1000,
    director: `Frank Iero`,
    description: `Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tom Skerritt`,
      `Sigourney Weaver`,
      `Veronica Cartwright`,
      `Harry Dean Stanton`,
      `John Hurt`,
      `Ian Holm`,
      `Yaphet Kotto`
    ],
  },
  {
    id: `what-we-do-in-the-shadows`,
    name: `Orlando`,
    genre: `Dramas`,
    year: 2011,
    image: `img/what-we-do-in-the-shadows.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    time: `2h 00m`,
    rating: 7.5,
    votes: 1000,
    director: `Ice Loched`,
    description: `Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there.`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tom Skerritt`,
      `Sigourney Weaver`,
      `Veronica Cartwright`,
      `Harry Dean Stanton`,
      `John Hurt`,
      `Ian Holm`,
      `Yaphet Kotto`
    ],
  },
  {
    id: `no-country-for-old-men`,
    name: `Snatch`,
    genre: `Dramas`,
    year: 2009,
    image: `img/no-country-for-old-men.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: `Jane Doe`,
        text: `Fantastic!`
      }
    ], starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`,
      `Hugo Weaving`,
      `Joe Pantoliano`
    ],
  }
];
