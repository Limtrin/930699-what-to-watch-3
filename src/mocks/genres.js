
const FILMS_GENRES = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];

export default FILMS_GENRES;
