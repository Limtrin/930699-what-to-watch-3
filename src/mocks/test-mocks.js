export const filmsList = [
  {
    id: `shutter-island`,
    name: `Shutter Island`,
    genre: `detective`,
    year: 2010,
    image: `img/shutter-island.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  },
  {
    id: `snatch`,
    name: `Snatch`,
    genre: `comedy`,
    year: 2009,
    image: `img/snatch.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  }
];

export const film = {
  id: `war-of-the-worlds`,
  name: `War of the Worlds`,
  genre: `action`,
  year: 2008,
  image: `img/war-of-the-worlds.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `2h 22m`,
  rating: 9.5,
  votes: 123,
  director: `Frank Spaskukotski`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
  reviews: [
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    }
  ],
  starring: [
    `Tim Robbins`,
    `Morgan Freeman`,
    `Bob Gunton`,
    `William Sadler`,
    `Clancy Brown`,
    `Gil Bellows`,
    `James Whitmore`
  ],
};
