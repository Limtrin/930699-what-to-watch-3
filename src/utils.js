export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const ratingTransition = (rating) => {

  let filmRating;
  switch (true) {
    case (rating <= 3):
      filmRating = `Bad`;
      break;
    case (rating <= 5):
      filmRating = `Normal`;
      break;
    case (rating <= 8):
      filmRating = `Good`;
      break;
    case (rating < 10):
      filmRating = `Very Good`;
      break;
    default:
      filmRating = `Awesome`;
  }
  return filmRating;
};
