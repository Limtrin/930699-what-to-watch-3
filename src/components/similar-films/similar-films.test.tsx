import * as React from "react";
import * as renderer from "react-test-renderer";
import SimilarFIlms from "./similar-films";
import {filmsList, film} from "../../mocks/test-mocks";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {noop} from "../../utils";

const mockStore = configureStore([]);

it(`Should render SimilarFilms component`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      genre: `All genres`,
      filmsList,
      filmsCurrent: filmsList
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <SimilarFIlms
            film={film}
            onFilmCardClickHandler={noop}
            onHeaderClickHandler={noop}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
