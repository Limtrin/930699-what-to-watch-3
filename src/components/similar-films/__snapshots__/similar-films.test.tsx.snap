// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SimilarFilms component 1`] = `
<div
  className="catalog__movies-list"
/>
`;
